core/server/src/main/java/alluxio/master/FaultTolerantAlluxioMaster.java

```diff
@@ -47,9 +48,10 @@ public FaultTolerantAlluxioMaster() {
 
     // Set up zookeeper specific functionality.
     try {
-      // InetSocketAddress.toString causes test issues, so build the string by hand
-      String zkName = NetworkAddressUtils.getConnectHost(ServiceType.MASTER_RPC) + ":"
-          + getRpcAddress().getPort();
+      String zkName = new MasterZooKeeperNode(
```

Comment:
What are the implications of this, with respect to backwards compatibility? Older zk names will be of a different format, so when updating the master, what happens? What happens with an old Alluxio client (and MasterInquireClient) and new master, and vice versa?